frame/multisig/src/lib.rs

```diff
@@ -600,12 +605,15 @@ impl<T: Trait> Module<T> {
 
 			// Just start the operation by recording it in storage.
 			let deposit = T::DepositBase::get() + T::DepositFactor::get() * threshold.into();
-			T::Currency::reserve(&who, deposit)?;
 
 			// Store the call if desired.
-			let stored = maybe_call
-				.filter(|_| store)
-				.map_or(false, |data| Self::store_call(who.clone(), &call_hash, data));
+			let stored = if let Some(data) = maybe_call.filter(|_| store) {
+				Self::store_call_and_reserve(who.clone(), &call_hash, data, deposit)?;
+				true
+			} else {
+				T::Currency::reserve(&who, deposit)?;
```

Comment:
We reserve anyway?

Reply:
This is the branch where the user did *not*  want to store (passed `store == false`) or the call was not yet present (`maybe_call == None`), so we do continue, but we only reserve the deposit for the `Multisigs::inser` below.
So IMO this code is correct as written.

Reply:
yes I see, thanks